include/seqan3/core/simd/detail/simd_algorithm_avx2.hpp

```diff
@@ -0,0 +1,152 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides specific algorithm implementations for AVX2 instruction set.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/core/simd/concept.hpp>
+#include <seqan3/core/simd/detail/builtin_simd_intrinsics.hpp>
+#include <seqan3/core/simd/simd_traits.hpp>
+
+//-----------------------------------------------------------------------------
+// forward declare avx2 simd algorithms that use avx2 intrinsics
+//-----------------------------------------------------------------------------
+
+namespace seqan3::detail
+{
+/*!\copydoc seqan3::simd::load
+ * \attention This is the implementation for AVX2 intrinsics.
+ */
+template <simd::simd_concept simd_t>
+constexpr simd_t load_avx2(void const * mem_addr);
+
+/*!\copydoc seqan3::simd::transpose
+ * \attention This is the implementation for AVX2 intrinsics.
+ */
+template <simd::simd_concept simd_t>
+inline void transpose_matrix_avx2(std::array<simd_t, simd_traits<simd_t>::length> & matrix);
+
+/*!\copydoc seqan3::detail::upcast_signed
+ * \attention This is the implementation for AVX2 intrinsics.
+ */
+template <simd::simd_concept target_simd_t, simd::simd_concept source_simd_t>
+constexpr target_simd_t upcast_signed_avx2(source_simd_t const & src);
+
+/*!\copydoc seqan3::detail::upcast_unsigned
+ * \attention This is the implementation for AVX2 intrinsics.
+ */
+template <simd::simd_concept target_simd_t, simd::simd_concept source_simd_t>
+constexpr target_simd_t upcast_unsigned_avx2(source_simd_t const & src);
+
+/*!\copydoc seqan3::detail::extract_halve
```

Comment:
```suggestion
/*!\copydoc seqan3::detail::extract_half
```

At some point it should be renamed...

Reply:
Next PR after this will rename it.